src/amo/reducers/reviews.js

```diff
@@ -313,15 +335,14 @@ export default function reviewsReducer(
     }
     case SET_REVIEW: {
       const { payload } = action;
-      return {
-        ...state,
-        byId: storeReviewObjects({ state, reviews: [payload] }),
-        byUserId: {
-          ...state.byUserId,
-          // This will trigger a refresh from the server.
-          [payload.userId]: undefined,
-        },
-      };
+      const review = createInternalReview(payload);
+
+      return addReviewInState({ state, review });
```

Comment:
Actually, yeah, this abstraction totally works 😃 I'm not sure why I thought it wouldn't.